app/src/main/java/com/github/se/polyfit/ui/viewModel/GraphViewModel.kt

```diff
@@ -0,0 +1,128 @@
+package com.github.se.polyfit.ui.viewModel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import co.yml.charts.common.model.Point
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+
+enum class SortPoints {
+  KCAL,
+  WEIGHT;
+
+  override fun toString(): String {
+    return this.name
+  }
+
+  companion object {
+    fun toList(): List<String> {
+      return values().map { it.toString() }
+    }
+  }
+}
+
+enum class SortDirection {
+  ASCENDING,
+  DESCENDING
+}
+
+class GraphViewModel : ViewModel() {
+  private val _sortPoints = MutableStateFlow(SortPoints.KCAL)
+
+  private val _sortDirection = MutableStateFlow(SortDirection.ASCENDING)
+
+  private val _searchText = MutableStateFlow("")
+  val searchText = _searchText.asStateFlow()
+
+  private val _isSearching = MutableStateFlow(false)
+  val isSearching = _isSearching.asStateFlow()
+
+  private val _graphData = MutableStateFlow(mockData)
+  val graphData =
```

Comment:
Small suggestion to make understanding what this variable mean could be to refactor the sorting into its own function.

Reply:
done !